server.go

```diff
@@ -1854,6 +1854,9 @@ func (sc *serverConn) newWriterAndRequestNoBody(st *stream) (*responseWriter, er
 }
 
 func writeResponseBody(rw *responseWriter, reqCtx *app.RequestContext) (err error) {
+	if !bodyAllowedForStatus(reqCtx.Response.StatusCode()) {
```

Comment:
There exists a scenario: if users really set a body with these status codes and they will not get any error from that and no body will be sent - which is a different way of process compare to the standard lib?

Reply:
rfc 也定义了 204和304 https://www.rfc-editor.org/rfc/rfc9110.html#name-204-no-content  不能包含body，即便设置了body也会被忽略。代理比如nginx也会忽略掉body部分，所以设置并没有太大意义。

Reply:
@rogerogers is correct, BUT actually i prefer letting the body set because most customized http servers allow so and customized versions of client also will allow so.

but i can live with the rfc standard.

@welkeyever, can you allow the pr for now and think later because not allowing the pull requests is worst than everything else you can think of for now.

we can discuss this AFTER implementation because right now, just setting 304 crashes.

Reply:
@rogerogers I think both are ok - if you give a reasonable reason for this change. 
Can you add this part of the change to the description of the pr？